packages/react-storage/src/components/StorageBrowser/Views/Controls/Summary.tsx

```diff
@@ -1,27 +1,100 @@
 import React from 'react';
+import { withBaseElementProps } from '@aws-amplify/ui-react-core/elements';
+
 import { StorageBrowserElements } from '../../context/elements';
+import { CLASS_BASE } from '../constants';
+
+const { Button, DefinitionDetail, DefinitionList, DefinitionTerm, View } =
+  StorageBrowserElements;
+
+const BLOCK_NAME = `${CLASS_BASE}__summary`;
+
+/* <Detail /> */
+const DetailContainer = withBaseElementProps(View, {
+  className: `${CLASS_BASE}__detail`,
+});
+
+const DetailLabel = withBaseElementProps(DefinitionTerm, {
+  className: `${CLASS_BASE}__detail__label`,
+});
+const DetailValue = withBaseElementProps(DefinitionDetail, {
+  className: `${CLASS_BASE}__detail__value`,
+});
+
+interface DetailProps {
+  label?: string;
+  value?: string;
+}
+interface Detail<T extends StorageBrowserElements = StorageBrowserElements> {
+  (props: DetailProps): React.JSX.Element;
+  Container: T['View'];
+  Label: T['DefinitionTerm'];
+  Value: T['DefinitionDetail'];
+}
 
-const { Button, Span, Text } = StorageBrowserElements;
+const Detail: Detail = (props: DetailProps) => {
+  const { label, value } = props;
+
+  return (
+    <DetailContainer>
+      <DetailLabel>{label}</DetailLabel>
+      <DetailValue>{value}</DetailValue>
+    </DetailContainer>
+  );
+};
+Detail.Container = DetailContainer;
+Detail.Label = DetailLabel;
+Detail.Value = DetailValue;
+
+const CompletedControl = ({
+  label = 'Completed',
+  value = '0/5',
+}: DetailProps) => <Detail label={label} value={value} />;
+
+const NotStartedControl = ({
+  label = 'Not started',
+  value = '0/5',
+}: DetailProps) => <Detail label={label} value={value} />;
+
+const FailedControl = ({ label = 'Failed', value = '0/5' }: DetailProps) => (
+  <Detail label={label} value={value} />
+);
+
+const CanceledControl = ({
+  label = 'Canceled',
+  value = '0/5',
+}: DetailProps) => <Detail label={label} value={value} />;
+
+/* <Details /> */
+
+const DetailsContainer = withBaseElementProps(DefinitionList, {
+  className: BLOCK_NAME,
+});
 
 interface Details<T extends StorageBrowserElements = StorageBrowserElements> {
   (): React.JSX.Element;
-  Container: T['Span'];
-  Destination: T['Text'];
-  Completed: T['Text'];
-  Failed: T['Text'];
-  Canceled: T['Text'];
-  NotStarted: T['Text'];
+  Container: T['DefinitionList'];
+  Completed: Detail<T>;
+  Failed: Detail<T>;
+  Canceled: Detail<T>;
+  NotStarted: Detail<T>;
 }
 
-const Details: Details = () => <>Hi</>;
-
-Details.Canceled = Text;
-Details.Completed = Text;
-Details.Container = Span;
-Details.Destination = Text;
-Details.Failed = Text;
-Details.NotStarted = Text;
+const Details: Details = () => (
+  <DetailsContainer>
+    <CompletedControl />
+    <CanceledControl />
+    <FailedControl />
+    <NotStartedControl />
+  </DetailsContainer>
+);
 
+Details.Container = DetailsContainer;
+Details.Canceled = CanceledControl;
```

Comment:
I thought I could make the individual Completed, Failed, etc controls and instance of Detail. But I'm having trouble with this type error when I try to assign them as properties on Details:

<img width="868" alt="Type '({ label, value, }: DetailProps) => Element' is missing the following properties from type 'Detail<StorageBrowserElements>': Container, Label, Value" src="https://github.com/user-attachments/assets/45a3248e-8049-4996-8afb-871efbd266a9">